primary/tests/integration_tests_configuration_api.rs

```diff
@@ -181,6 +182,61 @@ async fn test_new_network_info() {
     assert_eq!(Empty {}, actual_result);
 }
 
+#[tokio::test]
+async fn test_get_primary_address() {
```

Comment:
I believe you can benefit in this test by [Cluster](https://github.com/MystenLabs/narwhal/blob/main/test_utils/src/cluster.rs). It will allow you to easily spin up - even just one node - and perform this test without having to do this setup . A small addition might be needed so the cluster can run without consensus - I'll see whether I can have something quickly so you can use it here.

Reply:
@arun-koshy you can find the changes on [this PR](https://github.com/MystenLabs/narwhal/pull/599) . An example of how you could use the `Cluster` in this test can be found [here](https://github.com/MystenLabs/narwhal/pull/599/files#diff-23e4050e4099fea2c66148e23cfb98113529fa70502a78764b98494536b4c19dR39)

Reply:
This is awesome! Thanks Tasos 🤩

Reply:
Will wait for your PR to merge, to properly utilize your changes.